### ISSUE TITLE
Choice of Colors in Browser Coding Window in FreeCodeCamp Challenges for Accessibility for Color Blind

### ISSUE THREAD

**Issue by VincentDiCarlo** [freeCodeCamp/freeCodeCamp#8845]
#### Challenge Name

All
#### Issue Description

A way to change the background color of the code window from black to white, and the default code color from white to black. I am color impaired and have trouble seeing red characters on a black background.  I believe about 6% of all men have a similar condition.

In the screenshot below, I can make out the characters in red, but it is not easy.  A white background with either black or colored text would make a big difference.

I know that as a workaround I can use an external editor and past the results in the code window, but that might be a little daunting for some absolute novices.
#### Browser Information

- Browser Name, Version: 50.0.2661.102 m
- Operating System: Windows 7 Professional Services Pack 1
- Mobile, Desktop, or Tablet: Desktop
#### Your Code

``` js
// If relevant, paste all of your challenge code in here

```
#### Screenshot

![image](https://cloud.githubusercontent.com/assets/19619393/15631325/20a9b00a-2519-11e6-8597-545b10e49fef.png)

**Comment by erictleung** [comment]
@VincentDiCarlo thanks for the feedback. We strive to make the site as accessible as possible. The closest thing we have within the Free Code Camp site that might help is night mode. You can access this by typing `g`,`t`,`n` within the site. Let us know how that looks to you. Another solution is to look for extensions for your browser that enable color impaired friendly themes.

cc/ @FreeCodeCamp/issue-moderators

**Comment by hallaathrad** [comment]
Maybe we could come up with a high contrast mode, without a key shortcut, for it most likely would be a permanently turned-on feature for some people.

Do we have an a11y milestone on the works anywhere? Or should this be a separate thing?

**Comment by raisedadead** [comment]
We sure should have one a11y milestone! and +1 for a high contrast mode.

/cc @QuincyLarson

**Comment by QuincyLarson** [comment]
@raisedadead @hallaathrad wait - our default mode is very high contrast. 

![](https://www.evernote.com/l/Ar8sNrd2b4hBvpZNkIfkPH4UE5QSlELuOg0B/image.png)

Would this be a third mode?

**Comment by hallaathrad** [comment]
@QuincyLarson I'd think so, yes. The "night mode" is more of a "stop burning my eyes, it's late and they hurt", rather than a full-on high contrast (less colourful, a11y oriented one). As the op correctly states, a lot of people have trouble reading non-bright colours over dark backgrounds... I'd advocate for a bright grayscale over black backgrounds —very simple, not precisely pretty... but highly functional in these cases in which impairments are actually the main focus over mere aesthetics.

**Comment by raisedadead** [comment]
@QuincyLarson as the OP mentions they are having trouble with the code editor itself:

>  I am color impaired and have trouble seeing **red characters on a black background**. I believe about 6% of all men have a similar condition.

Refer screenshot in the OP.

**Comment by jcolsen** [comment]
I'm red/green color blind (the most common kind) and I think the Night Mode editor is the hardest one to read, expecially when typing and that highlighter kicks in since (depending on what you do) it outputs green text with a red highlight behind the text while typing. 

If I may suggest something along the lines of the Atom default syntax highlighter? It's not perfect, but it's a million times easier to read than both the current regular and especially the night mode editor. The editor and site on Codecademy is also a good example on what works.

As a side note: Readability in general can be quite challenging on most of the site that I've seen so far for the same reason; too much use of red/green as a combination as well as value of red/green.

For now I've recieved a tip to install something called GhostText so I can use my own editor instead. It's only a partial solution though as it doesn't change what's outside of the editor bit.

**Comment by y-o-r-i-s** [comment]
I would love it if there were more theme options, for the site, for the code editing window, etc. The same way that IDE's allow you to customize everything the way you like it.

I've only now found Night Mode; not colorblind, not coding through the night, I just prefer the colors. The default colors annoy and distract me, passively - I can feel a certain distaste in my brain after being on the site for a bit. Especially that darkish pink for <div> and <scripts> etc.

**Comment by itsabear** [comment]
So we should have options to change coloring syntax theme too? @y-o-r-i-s

**Comment by kent37** [comment]
I also have trouble reading colored text on a black background. A high-contrast setting which keeps the black background would be just as hard to read.
